internal/revproxy/auth.go

```diff
@@ -18,6 +18,25 @@ import (
 
 type AuthOption func(*Auth)
 
+type AuthMiddlewareProvider interface {
+	Middleware() echo.MiddlewareFunc
+}
+
+func NewAuthMiddlewareProvider(ignoreGitlabTokens bool, options ...AuthOption) (AuthMiddlewareProvider, error) {
```

Comment:
If I set `ignoreGitlabTokens` to true on a middleware that has nothing to do with GitLab, it will skip the auth entirely too, so the way the arguments are named here are an anti-pattern.